account-keys/src/account_keys.rs

```diff
@@ -475,6 +475,180 @@ impl AccountKey {
     }
 }
 
+/// View AccountKey, containing the view private key and the spend public key.
+#[derive(Clone, Message, Zeroize)]
+#[zeroize(drop)]
+pub struct ViewAccountKey {
+    /// Private key 'a' used for view-key matching.
+    #[prost(message, required, tag = "1")]
+    view_private_key: RistrettoPrivate,
+
+    /// Public key `B` used for generating Public Addresses.
+    #[prost(message, required, tag = "2")]
+    spend_public_key: RistrettoPublic,
+}
+
+// Note: Hash, Ord is implemented in terms of default_subaddress() because
+// we don't want comparisons to leak private key details over side-channels.
+impl Hash for ViewAccountKey {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.default_subaddress().hash(state)
+    }
+}
+
+impl Eq for ViewAccountKey {}
+
+impl PartialEq for ViewAccountKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.default_subaddress().eq(&other.default_subaddress())
+    }
+}
+
+impl PartialOrd for ViewAccountKey {
+    fn partial_cmp(&self, other: &ViewAccountKey) -> Option<Ordering> {
+        Some(self.default_subaddress().cmp(&other.default_subaddress()))
+    }
+}
+
+impl Ord for ViewAccountKey {
+    fn cmp(&self, other: &ViewAccountKey) -> Ordering {
+        self.default_subaddress().cmp(&other.default_subaddress())
+    }
+}
+
+impl From<&AccountKey> for ViewAccountKey {
+    fn from(account_key: &AccountKey) -> Self {
+        ViewAccountKey {
+            view_private_key: *account_key.view_private_key(),
+            spend_public_key: account_key.spend_private_key().into(),
+        }
+    }
+}
+
+impl ViewAccountKey {
+    /// A user's ViewAccountKey, without a fog service.
+    ///
+    /// # Arguments
+    /// * `view_private_key` - The user's private view key `a`.
+    /// * `spend_public_key` - The user's public spend key `B`.
+    #[inline]
+    pub fn new(view_private_key: RistrettoPrivate, spend_public_key: RistrettoPublic) -> Self {
+        Self {
+            view_private_key,
+            spend_public_key,
+        }
+    }
+
+    /// Get the view private key.
+    pub fn view_private_key(&self) -> &RistrettoPrivate {
+        &self.view_private_key
+    }
+
+    /// Get the spend public key.
+    pub fn spend_public_key(&self) -> &RistrettoPublic {
+        &self.spend_public_key
+    }
+
+    /// Create a view account key with random keys
+    pub fn random<T: RngCore + CryptoRng>(rng: &mut T) -> Self {
```

Comment:
Optional/Equivalent: Use `impl` instead of generics
```suggestion
    pub fn random(rng: &mut (impl RngCore + CryptoRng)) -> Self {
```